sqlx-sqlite/src/connection/serialized_database.rs

```diff
@@ -0,0 +1,32 @@
+#![allow(unsafe_code)]
+extern crate libsqlite3_sys as ffi;
+
+/// `SerializedDatabase` is a wrapper for a serialized database that is dynamically allocated by calling `sqlite3_serialize`.
+/// This RAII wrapper is necessary to deallocate the memory when it goes out of scope with `sqlite3_free`.
+#[derive(Debug)]
+pub struct SerializedDatabase {
+    data: *mut u8,
```

Comment:
This can be `NonNull<u8>` to make this struct eligible for null pointer optimization.